### PR TITLE
provided tests for Fruit Shop

### PR DESCRIPTION
**Pull request by yurii-dikalo**

### CODE REVIEW

File: src/test/java/core/basesyntax/strategy/operationhandler/impl/PurchaseOperationHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        assertEquals(expected, actual);
+    }
+
+    @Test (expected = RuntimeException.class)
+    public void handle_greaterPurchaseQuantity_isOk() {
+        transaction.setQuantity(GREATER_PURCHASE_QUANTITY);
+        purchaseOperationHandler.handle(transaction);
+        int actual = FruitStorage.fruitStorage.get(APPLE);
+        int expected = QUANTITY - PURCHASE_QUANTITY;
+        assertEquals(expected, actual);
```
Comment (lysytskapolina) [inline src/test/java/core/basesyntax/strategy/operationhandler/impl/PurchaseOperationHandlerTest.java:41]: Aren't these lines redundant after you're expecting RuntimeException?

File: src/test/java/core/basesyntax/strategy/operationhandler/impl/SupplyOperationHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.strategy.operationhandler.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class SupplyOperationHandlerTest {
```
Comment (lysytskapolina) [inline src/test/java/core/basesyntax/strategy/operationhandler/impl/SupplyOperationHandlerTest.java:10]: (suggestion only) You also can add test in which supply quantity is 0, because then this operation is redundant, for me) It also relates to ReturnOperationHandlerTest.

File: src/test/java/core/basesyntax/strategy/operationhandler/impl/SupplyOperationHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    private static final String APPLE = "apple";
+    private static final int APPLE_QUANTITY = 50;
+    private static final int APPLE_RETURN_QUANTITY = 30;
```
Comment (lysytskapolina) [inline src/test/java/core/basesyntax/strategy/operationhandler/impl/SupplyOperationHandlerTest.java:13]:
```suggestion
    private static final int APPLE_SUPPLY_QUANTITY = 30;
```

File: src/test/java/core/basesyntax/strategy/operationhandler/impl/BalanceOperationHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class BalanceOperationHandlerTest {
```
Comment (lysytskapolina) [inline src/test/java/core/basesyntax/strategy/operationhandler/impl/BalanceOperationHandlerTest.java:12]: (suggestion only) You can also add test in which balance is negative number. For me, it's really bad for fruit shop's owner)

File: src/test/java/core/basesyntax/service/impl/ProcessDataServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import core.basesyntax.strategy.operationhandler.impl.BalanceOperationHandler;
+import core.basesyntax.strategy.operationhandler.impl.PurchaseOperationHandler;
+import core.basesyntax.strategy.operationhandler.impl.ReturnOperationHandler;
+import core.basesyntax.strategy.operationhandler.impl.SupplyOperationHandler;
+import java.util.List;
+import java.util.Map;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ProcessDataServiceImplTest {
```
Comment (lysytskapolina) [inline src/test/java/core/basesyntax/service/impl/ProcessDataServiceImplTest.java:19]: (suggestion only) You can also add test in which data will be null)

File: src/test/java/core/basesyntax/service/impl/FruitTransactionParseServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ParseService;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitTransactionParseServiceTest {
```
Comment (lysytskapolina) [inline src/test/java/core/basesyntax/service/impl/FruitTransactionParseServiceTest.java:10]: (suggestion only) You can also add test in which data will be null)

File: src/test/java/core/basesyntax/service/impl/FileWriterServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.FileWriterService;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FileWriterServiceImplTest {
```
Comment (lysytskapolina) [inline src/test/java/core/basesyntax/service/impl/FileWriterServiceImplTest.java:7]: (suggestion only) You can also add test in which path will be null)

File: src/test/java/core/basesyntax/service/impl/FileReaderServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+import core.basesyntax.service.FileReaderService;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FileReaderServiceImplTest {
```
Comment (lysytskapolina) [inline src/test/java/core/basesyntax/service/impl/FileReaderServiceImplTest.java:13]: (suggestion only) You can also add test in which path will be null)

**Review by lysytskapolina — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/service/impl/FileReaderServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    private static final String VALID_INPUT_PATH = "src/test/resources/input_test_file.csv";
+    private static final String INVALID_INPUT_PATH = "";
```
[the remark this answers is not in the document]
Comment (yurii-dikalo) [reply src/test/java/core/basesyntax/service/impl/FileReaderServiceImplTest.java:15]: I don't know if I have to use it, but it just looks more readable to me

File: src/test/java/core/basesyntax/strategy/operationhandler/impl/BalanceOperationHandlerTest.java (hunk 4 above)
Comment (yurii-dikalo) [reply src/test/java/core/basesyntax/strategy/operationhandler/impl/BalanceOperationHandlerTest.java:12]: I am testing this case in FruitTransactionParseService. there's no way that transactions with negative quantity will be passed to this stage

File: src/test/java/core/basesyntax/strategy/operationhandler/impl/SupplyOperationHandlerTest.java (hunk 2 above)
Comment (yurii-dikalo) [reply src/test/java/core/basesyntax/strategy/operationhandler/impl/SupplyOperationHandlerTest.java:10]: I don't know why quantity is not ok to be 0, because then you don't do anything with it. But maybe I could have added check in FruitTransactionParseService, so no transaction with int quantity = 0 will even be parsed. Is it OK?

File: src/test/java/core/basesyntax/strategy/operationhandler/impl/BalanceOperationHandlerTest.java (hunk 4 above)
Comment (lysytskapolina) [reply src/test/java/core/basesyntax/strategy/operationhandler/impl/BalanceOperationHandlerTest.java:12]: Sorry, I was inattentive)

File: src/test/java/core/basesyntax/dao/impl/FruitDaoImplTest.java [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+public class FruitDaoImplTest {
+    private static final String APPLE = "apple";
+    private static final String BANANA = "banana";
+    private static final int DEFAULT_QUANTITY = 50;
+    private static final int NEW_QUANTITY = 100;
+    private static FruitDao fruitDao;
+
+    @BeforeClass
+    public static void beforeClass() {
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/dao/impl/FruitDaoImplTest.java:24]:
```suggestion
    public static void init() {
```

File: src/test/java/core/basesyntax/dao/impl/FruitDaoImplTest.java [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+        fruitDao = new FruitDaoImpl();
+    }
+
+    @Before
+    public void beforeEach() {
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/dao/impl/FruitDaoImplTest.java:29]:
```suggestion
    public void fillStorage() {
```

File: src/test/java/core/basesyntax/dao/impl/FruitDaoImplTest.java [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+        FruitStorage.fruitStorage.put(APPLE, DEFAULT_QUANTITY);
+    }
+
+    @After
+    public void afterEach() {
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/dao/impl/FruitDaoImplTest.java:34]:
```suggestion
    public void clearStorage() {
```
 the similar naming should be in other test classes, don't forget, name of the methods should start with verb

File: src/test/java/core/basesyntax/service/impl/FileWriterServiceImplTest.java [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FileWriterServiceImplTest {
+    private static final String VALID_OUTPUT_PATH = "src/test/resources/output_test_file.csv";
+    private static final String INVALID_OUTPUT_PATH = "";
+    private static final String REPORT = "fruit,quantity" + System.lineSeparator()
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/service/impl/FileWriterServiceImplTest.java:15]: make it local

File: src/test/java/core/basesyntax/service/impl/FileWriterServiceImplTest.java [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    @Test (expected = RuntimeException.class)
+    public void writeToFile_writeToNonExistingFile_isNotOk() {
+        fileWriterService.writeToFile(INVALID_OUTPUT_PATH, REPORT);
+    }
+
+    @Test (expected = NullPointerException.class)
+    public void writeToFile_nullFilePath_isNotOk() {
+        fileWriterService.writeToFile(null, REPORT);
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/service/impl/FileWriterServiceImplTest.java:39]:
```suggestion
        fileWriterService.writeToFile(null, "");
```

File: src/test/java/core/basesyntax/service/impl/FruitTransactionParseServiceTest.java [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+    private static final String VALID_DATA = "b,banana,50";
+    private static final String INVALID_DATA = "b,banana,-50";
+    private static final String WRONG_DELIMITER_DATA = "b banana 50";
+    private static final String FRUIT = "banana";
+    private static final FruitTransaction.Operation OPERATION = FruitTransaction.Operation.BALANCE;
+    private static final int QUANTITY = 50;
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/service/impl/FruitTransactionParseServiceTest.java:16]: make them local

File: src/test/java/core/basesyntax/strategy/operationhandler/impl/BalanceOperationHandlerTest.java [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+    private static final String APPLE = "apple";
+    private static final String BANANA = "banana";
+    private static final int APPLE_QUANTITY = 40;
+    private static final int BANANA_QUANTITY = 50;
+    private static FruitTransaction transaction;
+    private static BalanceOperationHandler balanceOperationHandler;
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/strategy/operationhandler/impl/BalanceOperationHandlerTest.java:18]:
```suggestion
    private static OperationHandler balanceOperationHandler;
```
the same for other implementations test classes

File: src/test/java/core/basesyntax/strategy/operationhandler/impl/BalanceOperationHandlerTest.java [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+    public static void beforeClass() {
+        transaction = new FruitTransaction();
+        balanceOperationHandler = new BalanceOperationHandler();
+        FruitStorage.fruitStorage.put(APPLE, APPLE_QUANTITY);
+        transaction.setFruit(BANANA);
+        transaction.setQuantity(BANANA_QUANTITY);
+    }
+
+    @Test
+    public void handle_balance_isOk() {
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/strategy/operationhandler/impl/BalanceOperationHandlerTest.java:30]:
```suggestion
    public void handle_isOk() {
```

**Review by Nick97-git — CHANGES_REQUESTED** [review]
left comments

**Review by Nick97-git — APPROVED** [review]
